src/main/java/gregtech/common/items/behaviors/spray/DurabilitySprayBehavior.java

```diff
@@ -0,0 +1,129 @@
+package gregtech.common.items.behaviors.spray;
+
+import gregtech.api.items.metaitem.stats.IItemDurabilityManager;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.GradientUtil;
+
+import net.minecraft.client.resources.I18n;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.Constants;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.util.List;
+
+public class DurabilitySprayBehavior extends AbstractSprayBehavior implements IItemDurabilityManager {
+
+    public static final String NBT_KEY = "GT.UsesLeft";
+
+    @Nullable
+    private final EnumDyeColor color;
+    @NotNull
+    private final Pair<Color, Color> durabilityBarColors;
+    @NotNull
+    private final ItemStack replacementStack;
+    public final int maxUses;
+
+    public DurabilitySprayBehavior(@NotNull ItemStack replacementStack, int maxUses, @Nullable EnumDyeColor color) {
+        this.color = color;
+        this.replacementStack = replacementStack;
+        this.maxUses = maxUses;
+        this.durabilityBarColors = GradientUtil.getGradient(color == null ? 0x969696 : color.colorValue, 10);
+    }
+
+    public DurabilitySprayBehavior(int maxUses, @Nullable EnumDyeColor color) {
+        this(ItemStack.EMPTY, maxUses, color);
+    }
+
+    @Override
+    public boolean canSpray(@NotNull ItemStack stack) {
+        return getUsesLeft(stack) > 0;
+    }
+
+    @Override
+    public void onSpray(@NotNull EntityPlayer player, @NotNull ItemStack sprayCan) {
+        if (player.capabilities.isCreativeMode) return;
+
+        if (damageCan(sprayCan)) {
+            if (replacementStack.isEmpty()) {
+                // If replacement stack is empty, just shrink resulting stack
+                sprayCan.shrink(1);
+            } else {
+                // Update held item to replacement stack
+                sprayCan.setItemDamage(replacementStack.getItemDamage());
+                // Clear NBT from old can
+                sprayCan.setTagCompound(new NBTTagCompound());
```

Comment:
This needs to be set to `null` I believe, not a new tag, for it to actually clear the NBT tag.